tilelang/env.py

```
@@ -253,12 +254,21 @@ class Environment:
         else:
-            NVSHMEM_SRC = os.path.join(
-                os.path.dirname(os.path.abspath(__file__)), "..", "3rdparty", "nvshmem_src")
-        NVSHMEM_INCLUDE_DIR: str = NVSHMEM_SRC + "/build/src/include"
-        NVSHMEM_LIB_PATH: str = NVSHMEM_SRC + "/build/src/lib"
+            # installed from wheel
+            from importlib.metadata import distribution
+
+            root = pathlib.Path(distribution("nvidia-nvshmem-cu12").locate_file(""))
+            nvshmem_wheel_path = root / "nvidia" / "nvshmem"
+            logger.warning(NVSHMEM_USE_DEFAULT_WHEEL_MESSAGE % nvshmem_wheel_path)
+            assert nvshmem_wheel_path.exists(), f"NVSHMEM wheel path does not exist"
```

Ruff (0.14.11)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)